lib/stm32/f4/rcc.c

```diff
@@ -578,13 +578,18 @@ void rcc_set_main_pll_hsi(uint32_t pllm, uint32_t plln, uint32_t pllp,
 }
 
 void rcc_set_main_pll_hse(uint32_t pllm, uint32_t plln, uint32_t pllp,
-			  uint32_t pllq)
+			  uint32_t pllq, uint32_t pllr)
 {
-	RCC_PLLCFGR = (pllm << RCC_PLLCFGR_PLLM_SHIFT) |
+	uint32_t reg32 = RCC_PLLCFGR & RCC_PLLCFGR_NO_R_RESERVED;
+	if (pllr != 0) {
+		reg32 &= RCC_PLLCFGR_R_RESERVED;
+	}
```

Comment:
given that you mask it with both anyway, it seems unecessary to have the full masks.  Could you add documentation to this function to explain the special use of the pllr param too please.  Also, is  there any particular reason you didn't add it to the hsi function?